src/main/java/htsjdk/tribble/util/URLHelperFactory.java

```diff
@@ -0,0 +1,20 @@
+package htsjdk.tribble.util;
+
+import java.net.URL;
+
+/**
+ * A factory for creating {@link  URLHelper} instances.  The factory contains a single function
+ * @see #getHelper(URL) which should return a <code>URLHelper</code> instance for the given URL.
+ */
+public interface URLHelperFactory {
+
+    /**
+     * @param url
+     * @return a {@link URLHelper} object for the given URL
+     * @throws
+     */
+    default URLHelper getHelper(URL url)   {
+        return new RemoteURLHelper(url);
```

Comment:
I would probably not make this a default method, since we probably want to require that anyone who implements this to actually override this method.  If this is non-default it lets us easily provide a lambda in place of a factory class as well since this would be a functional interface.

Reply:
Done